lms/djangoapps/grades/management/commands/compute_grades.py

```diff
@@ -103,11 +111,16 @@ def _get_course_keys(self, options):
         """
         Return a list of courses that need scores computed.
         """
-        courses_mode = get_mutually_exclusive_required_option(options, 'courses', 'all_courses')
-        if courses_mode == 'all_courses':
-            course_keys = [course.id for course in modulestore().get_course_summaries()]
+        if options.get('from_settings'):
+            if options.get('courses') or options.get('all_courses'):
+                raise CommandError('Cannot specify both --from_settings and either --courses or --all_courses')
```

Comment:
Save some logic by adding `'from_settings'` to the call to `get_mututally_exclusive_required_option()`.